### PR TITLE
[microsoft_dhcp] changes/fix to set values for source and the right values for host; #…

### LINKED ISSUE #7630: [microsoft_dhcp] Wrong field name usage in processing conflict with the content host.name
## Pitfall:
The actuall processing pipeline `packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/dhcp.yml` is processing the fields in wrong translation of context for the content of host.name, host.mac, host.domain (eg. for the hole part Field Object `host`).

## Result:
If you use the integration microsoft_dhcp as it is the Field Object `host` will be populated with `host.name, host.ip, host.domain` but those values are related to the Field Object `source` (with the source of the dhcp request). This raises the count of integrated host into a wrong number.

### Addition: Quote from the ECS for `host`:
`ECS host.* fields should be populated with details about the host on which the event happened, or from which the measurement was taken. Host types include hardware, virtual machines, Docker containers, and Kubernetes nodes.`
[ECS-Host](https://www.elastic.co/guide/en/ecs/current/ecs-host.html)

### Addition: Quote from the ECS for `source`:
`Source fields capture details about the sender of a network exchange/packet. These fields are populated from a network event, packet, or other event containing details of a network transaction.

Source fields are usually populated in conjunction with destination fields. The source and destination fields are considered the baseline and should always be filled if an event contains source and destination details from a network transaction. If the event also contains identification of the client and server roles, then the client and server fields should also be populated.`
[ECS-Source](https://www.elastic.co/guide/en/ecs/current/ecs-source.html)

## Solution and explanation:
Instead of setting the values from the CSV fields for ip,domain and MAC in the Field Object `host` the should be set in `source`. This is the right context cause those values are representing the sender of the request as a part of that network transaction.

### The test document is here which is a valid example: 
https://github.com/elastic/integrations/blob/main/packages/microsoft_dhcp/data_stream/log/sample_event.json

#### Line description of example how the CVS should be translated as followed:
`event.code,date,time,message,source.ip,source.domain,_tmp_.source.mac,......`

`source` = Requestor of DHCP  AND NOT `host`

I will submit a PR as proposal and hope that it is satisfied enough so that the context is right and ECS is respected.

### PR DESCRIPTION
**Pull request by xtruthx**
…7630

## Type of change
- Bug
- Enhancement

## What does this PR do?
Explainend in #7630 

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist
- [x] Is the handling for the field object `host` complete and valid (not sure actually how they created normally)
- [x] Not sure about the test *-expected.json i adapt the non DHCPv6 as example like i think it should bee

## Related issues
- Closes #7630 

Pinging @elastic/integrations (Team:Integrations)

### CODE REVIEW

**Comment by xtruthx** [comment]
Can some one look into this please? If i am wrong with my thoughts and the thoughts behind the merge request please let me know that. Ping @elastic/security-external-integrations

File: packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json [hunk 1]
```diff
@@ -80,12 +80,12 @@
...
             },
-            "host": {
-                "domain": "test.com",
+            "source": {
+                "domain": "057182593757.test.com",
                 "ip": [
                     "172.28.43.169"
                 ],
-                "name": "057182593757.test.com"
+                "address": "057182593757.test.com"
```
Comment (efd6) [inline packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:88]: The order of these fields looks odd (I would expect them to be in lexical order). Was the file manually edited?

File: packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -46,6 +46,26 @@ processors:
...
+      copy_from: observer.mac
+  - set:
+      field: host.name
+      override: false
+      copy_from: agent.name
+  - grok:
+      field: host.name
+      if: 'ctx.host?.name != null && ctx.host.name.contains(".")'
+      patterns:
+        - "%{HOSTNAME}\\.%{GREEDYDATA:host.domain}"
```
Comment (efd6) [inline packages/microsoft_dhcp/data_stream/log/elasticsearch/ingest_pipeline/default.yml:61]:
```suggestion
        - '%{HOSTNAME}\.%{GREEDYDATA:host.domain}'
```

File: packages/microsoft_dhcp/data_stream/log/fields/fields.yml [hunk 3]
```diff
@@ -64,4 +64,4 @@
     - name: subnet_prefix
       type: keyword
       description: |
-        The number of bits for the subnet prefix.
+        The number of bits for the subnet prefix.
```
Comment (efd6) [inline packages/microsoft_dhcp/data_stream/log/fields/fields.yml:67]: Please revert this change.

**Review by efd6 — COMMENTED** [review]
This seems reasonable. Please resolve the conflict and I will run the test build.

File: packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json (hunk 1 above)
Comment (xtruthx) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:88]: sorry i just substitute in "VIM" you are right.

File: packages/microsoft_dhcp/data_stream/log/fields/fields.yml (hunk 3 above)
Comment (xtruthx) [reply packages/microsoft_dhcp/data_stream/log/fields/fields.yml]: fixed with latest commit

**Comment by efd6** [comment]
Can you also resolve the conflict in the changelog (and ensure the manifest matches the resolution).

**Comment by xtruthx** [comment]
Is there still something to do? did i miss something?

**Comment by efd6** [comment]
Please run `elastic-package build`.

**Comment by xtruthx** [comment]
i did a rebase with latest changes from main. I did tests i build the packages and i did a test as described here:
https://github.com/elastic/integrations/blob/main/docs/developer_workflow_design_build_test_integration.md#steps

**Comment by efd6** [comment]
@xtruthx Please check that the changes that I've made match your expected semantics.

To answer your question, yes, the agent adds those fields in automatically. They don't need to be added into the tests, but the pipeline does need to be robust to their absence. These are the changes that I've made.

**Comment by xtruthx** [comment]
Thanks @efd6 . That looks great.
 Ok I see, so it is better to work with "ignore_empty_value".
Also that the description of the fields must match with the ECS I got now.

Sorry for the hopler and thanks for the patience. In the next post I know now how to do it:) 😄

File: packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json [hunk 4]
```diff
@@ -751,6 +711,10 @@
                     }
                 }
             },
+            "source": {
+                "ip": "192.168.10.40",
+                "mac": "65-34-45-37-2C-13-24-34-34-2A-38-13-37-30-2B-56-6C-61-6D-31"
```
Comment (efd6) [inline packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:716]: @xtruthx Sorry, I meant to ask this before, but forgot. Is this really a MAC? I've seen docs somewhere (don't recall where) that had an extended MAC+other on windows. Is this the case here?

Comment (xtruthx) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:716]: Not an issue. I overlooked that, too. But it seems that here the MAC address is added IMHO: This happens especially when DHCP policies are rolled out via GPO (not sure).

Microsoft DHCP uses the MAC address as client ID. Here it is then the client ID as a unique identifier.
The only good thing I could find is this:
https://superuser.com/questions/1552917/36-chars-mac-addresses-on-windows-server-2012-dhcp-server

I don't have any idea for a suggestion what to do.
Maybe check if it is a correct MAC and if there is no 12digit hexnum here, maybe cut out the last 12digit? If my above assumption is correct.

Comment (xtruthx) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:716]: This is a DHC-Server configuration option to use [DHCPv4 client-identifier](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-dhcpm/a9d6a415-4de6-4058-befe-ca2e42764485#gt_195d8585-7de6-4574-9abf-9db104aaa40c) or MA address only.

In DHCPv6 there is this the [DHCPv6 client-identifier/DUID](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-dhcpm/a9d6a415-4de6-4058-befe-ca2e42764485#gt_5c7b5d18-30dd-40fd-ac3e-a323f5663f51)

The Option would be in [Set-DhcpServerv4Reservation](https://learn.microsoft.com/en-us/powershell/module/dhcpserver/set-dhcpserverv4reservation?view=windowsserver2022-ps#-clientid)

should this be considered and handled in the package or is this something you write in the documentation and leave to the user's responsibility? But with the second option, the source.mac field might be filled with an incorrect value.

Comment (efd6) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:716]: Thanks, that does ring a bell.

For the moment I think we leave it up to user discretion, but I will raise it.

Would you mind resolving the conflict. Then I think this is probably good to go.

Comment (xtruthx) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:716]: done or?

Comment (efd6) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:716]: Timezones.

Also, for future spelunkers, [this](https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/dd183591(v=ws.10)) states that the field is a MAC.

Comment (xtruthx) [reply packages/microsoft_dhcp/data_stream/log/_dev/test/pipeline/test-log.log-expected.json:716]: yes timezones. I was aware. Thx

**Comment by efd6** [comment]
The deconfliction need to have an update to the manifest as well; this is an irritation with how merge conflict get resolved if you do it purely in a merge conflict editor.

**Comment by xtruthx** [comment]
> The deconfliction needs to have an update to the manifest as well; this is an irritation with how merge conflict get resolved if you do it purely in a merge conflict editor.

Done or sorry i forgot the manifest. Now i hope it looks good. Time hits me now.

**Review by efd6 — APPROVED** [review]
Thanks

**Comment by efd6** [comment]
@xtruthx I'm just wondering whether client wouldn't be better than source; this is the terminology used in the MS documentation.

**Comment by xtruthx** [comment]
@efd6 I first thought about putting it into the ECS object 'client'. But then I thought it might be better to use 'source' so that it is also used correctly in the SIEM/Security in the Network overview. I am not sure if the field 'client.source' is also asked. I'll have a look at it again in a moment.

**Comment by efd6** [comment]
Thanks.

**Comment by xtruthx** [comment]
@efd6 as it thank last 

> Thanks.

@efd6 i think that means yes please do the change and i did so.

When building the integration I got the following validation error:
```build the package
README.md file rendered: /home/daniel/git/elastic/integrations-fork/packages/microsoft_dhcp/docs/README.md
2023/11/02 06:23:30  INFO License text found in "/home/daniel/git/elastic/integrations-fork/LICENSE.txt" will be included in package
2023/11/02 06:23:30  INFO Skipped errors: found 1 validation error:
   1. conditions.kibana.version must be ^8.10.0 or greater to include saved object tags file: kibana/tags.yml (SVR00005)
Package built: /home/daniel/git/elastic/integrations-fork/build/packages/microsoft_dhcp-1.23.0.zip
Done
```
I think it makes sense to act accordingly and I have set the kibana version to ^8..10.0 only.

I hope now the merge is possible soon

**Comment by efd6** [comment]
> I think it makes sense to act accordingly and I have set the kibana version to ^8..10.0 only.

We'd rather this wasn't done since it cuts off users. The validation error really should be noted as a warning rather than an error as it is not something that blocks the build.

Please revert that line change.

**Comment by efd6** [comment]
We've had a discussion; please revert the source to client change.

**Comment by xtruthx** [comment]
sorry reverted

**Review by efd6 — APPROVED** [review]
Thanks
